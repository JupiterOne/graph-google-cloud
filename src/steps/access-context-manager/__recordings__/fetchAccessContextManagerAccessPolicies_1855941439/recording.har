{
  "log": {
    "_recordingName": "fetchAccessContextManagerAccessPolicies",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 300,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 572,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 572,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 02 Jun 2021 14:22:23 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-02T14:22:23.123Z",
        "time": 212,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 212
        }
      },
      {
        "_id": "e7ae7a9d3285d95d4330a7c239aa61b7",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.2 gl-node/14.15.4 auth/7.1.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.2 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "accesscontextmanager.googleapis.com"
            }
          ],
          "headersSize": 581,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "parent",
              "value": "organizations/958457776463"
            }
          ],
          "url": "https://accesscontextmanager.googleapis.com/v1/accessPolicies?parent=organizations%2F958457776463"
        },
        "response": {
          "bodySize": 369,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 369,
            "text": "{\"accessPolicies\":[{\"name\":\"accessPolicies/368605946207\",\"parent\":\"organizations/958457776463\",\"title\":\"default policy\",\"etag\":\"64186ca0c82b01b4\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 02 Jun 2021 14:22:24 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-02T14:22:23.374Z",
        "time": 829,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 829
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
