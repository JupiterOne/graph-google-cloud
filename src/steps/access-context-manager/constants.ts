export const STEP_ACCESS_CONTEXT_MANAGER_ACCESS_POLICIES =
  'fetch-access-context-manager-access-policies';
export const STEP_ACCESS_CONTEXT_MANAGER_ACCESS_LEVELS =
  'fetch-access-context-manager-access-levels';
export const STEP_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETERS =
  'fetch-access-context-manager-service-perimeters';

// Ideas: AccessPolicy, Policy
// also it HAS access_levels and service_perimeters too (maybe a Group too)
export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_ACCESS_POLICY = 'AccessPolicy';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_ACCESS_POLICY =
  'google_access_context_manager_access_policy';

// Ideas: Config, Requirement, Rule, Ruleset, ...
export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_ACCESS_LEVEL = 'Ruleset';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_ACCESS_LEVEL =
  'google_access_context_manager_access_level';

// Ideas: Config, Ruleset
export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER =
  'Configuration';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER =
  'google_access_context_manager_service_perimeter';

export const RELATIONSHIP_TYPE_ACCESS_POLICY_HAS_ACCESS_LEVEL =
  'google_access_context_manager_access_policy_has_level';
export const RELATIONSHIP_TYPE_ACCESS_POLICY_HAS_SERVICE_PERIMETER =
  'google_access_context_manager_access_policy_has_service_perimeter';
export const RELATIONSHIP_TYPE_SERVICE_PERIMETER_PROTECTS_PROJECT =
  'google_access_context_manager_service_perimeter_protects_project';
