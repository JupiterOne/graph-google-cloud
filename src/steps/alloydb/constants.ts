export const STEP_ALLOYDB_POSTGRE_SQL_SERVICE = 'create-postgre-sql-service';
export const STEP_ALLOYDB_POSTGRE_SQL_CLUSTER =
  'fetch-alloydb-postgre-sql-cluster';
export const STEP_ALLOYDB_POSTGRE_SQL_INSTANCE =
  'fetch-alloydb-postgre-sql-instance';
export const STEP_ALLOYDB_POSTGRE_SQL_CONNECTION =
  'fetch-alloydb-postgre-sql-connection';
export const STEP_ALLOYDB_POSTGRE_SQL_BACKUP =
  'fetch-alloydb-postgre-sql-backup';

// Entity type
export const ENTITY_TYPE_ALLOYDB_POSTGRE_SQL_SERVICE = 'google_cloud_alloydb';
export const ENTITY_CLASS_ALLOYDB_POSTGRE_SQL_SERVICE = ['Service'];

export const ENTITY_TYPE_POSTGRE_SQL_ALLOYDB_CLUSTER =
  'google_cloud_alloydb_cluster';
export const ENTITY_CLASS_POSTGRE_SQL_CLUSTER = [
  'Database',
  'DataStore',
  'Cluster',
];

export const ENTITY_TYPE_POSTGRE_SQL_ALLOYDB_INSTANCE =
  'google_cloud_alloydb_instance';
export const ENTITY_CLASS_POSTGRE_SQL_INSTANCE = [
  'Database',
  'DataStore',
  'Host',
];

export const ENTITY_TYPE_POSTGRE_SQL_CONNECTION =
  'google_cloud_alloydb_connection';
export const ENTITY_CLASS_POSTGRE_SQL_CONNECTION = ['Network'];

export const ENTITY_TYPE_POSTGRE_SQL_BACKUP = 'google_cloud_alloydb_backup';
export const ENTITY_CLASS_POSTGRE_SQL_BACKUP = ['Backup'];

// Relationships
export const STEP_PROJECT_HAS_ALLOYDB_SERVICE_RELATIONSHIP =
  'build-project-alloydb-service-relationship';
export const STEP_PROJECT_HAS_ALLOYDB_CLUSTER_RELATIONSHIP =
  'build-project-has-alloydb-cluster-relationship';

export const STEP_ALLOYDB_INSTANCE_USES_CLUSTER_RELATIONSHIP =
  'build-alloydb-instance-uses-cluster-relationship';

export const STEP_ALLOYDB_CLUSTER_HAS_BACKUP_RELATIONSHIP =
  'build-alloydb-cluster-has-backup-relationship';

export const STEP_ALLOYDB_INSTANCE_HAS_CONNECTION_RELATIONSHIP =
  'build-alloydb-instance-has-connection-relationship';

export const STEP_ALLOYDB_CLUSTER_USES_KMS_KEY_RELATIONSHIP =
  'build-alloydb-cluster-uses-kms-key-relationship';

export const STEP_USER_ASSIGNED_ALLOYDB_CLUSTER_RELATIONSHIP =
  'build-user-asssigned-alloydb-cluster-relatipnship';

export const RELATIONSHIP_TYPE_PROJECT_HAS_ALLOYDB_SERVICE =
  'google_cloud_project_has_alloydb';
export const RELATIONSHIP_TYPE_PROJECT_HAS_ALLOYDB_CLUSTER =
  'google_cloud_project_has_alloydb_cluster';
export const RELATIONSHIP_TYPE_ALLOYDB_INSTANCE_USES_CLUSTER =
  'google_cloud_alloydb_instance_uses_cluster';
export const RELATIONSHIP_TYPE_ALLOYDB_CLUSTER_HAS_BACKUP =
  'google_cloud_alloydb_cluster_has_backup';
export const RELATIONSHIP_TYPE_STEP_ALLOYDB_INSTANCE_HAS_CONNECTION_BACKUP =
  'google_cloud_alloydb_instance_has_connection';
export const RELATIONSHIP_TYPE_ALLOYDB_CLUSTER_USES_KMS_KEY =
  'google_cloud_alloydb_cluster_uses_kms_crypto_key';
export const RELATIONSHIP_TYPE_USER_ASSIGNED_ALLOYDB_CLUSTER =
  'google_user_assigned_cloud_alloydb_cluster';

// IngestionSources
export const IngestionSources = {
  ALLOYDB_POSTGRE_SQL_SERVICE: 'postgre-sql-service',
  ALLOYDB_POSTGRE_SQL_CLUSTER: 'alloydb-postgre-sql-cluster',
  ALLOYDB_POSTGRE_SQL_INSTANCE: 'alloydb-postgre-sql-instance',
  ALLOYDB_POSTGRE_SQL_CONNECTION: 'alloydb-postgre-sql-connection',
  ALLOYDB_POSTGRE_SQL_BACKUP: 'alloydb-postgre-sql-backup',
  PROJECT_HAS_ALLOYDB_SERVICE_RELATIONSHIP:
    'project-has-alloydb-service-relationship',
  PROJECT_HAS_ALLOYDB_CLUSTER_RELATIONSHIP:
    'project-has-alloydb-cluster-relationship',
  ALLOYDB_INSTANCE_USES_CLUSTER_RELATIONSHIP:
    'alloydb-instance-uses-cluster-relationship',
  ALLOYDB_CLUSTER_HAS_BACKUP_RELATIONSHIP:
    'alloydb-cluster-has-backup-relationship',
  ALLOYDB_INSTANCE_HAS_CONNECTION_RELATIONSHIP:
    'alloydb-instance-has-connection-relationship',
  ALLOYDB_CLUSTER_USES_KMS_KEY_RELATIONSHIP:
    'alloydb-cluster-uses-kms-key-relationship',
  GOOGLE_USER_ASSIGNED_ALLOYDB_CLUSTER_RELATIONSHIP:
    'google-user-assigned-alloydb-cluster-relationship',
};

// IngestionSources Configs
export const AlloyDBIngestionConfig = {
  [IngestionSources.ALLOYDB_POSTGRE_SQL_SERVICE]: {
    title: 'AlloyDB for Postgre SQL service',
    description: 'AlloyDB for Postgre SQL service.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_POSTGRE_SQL_CLUSTER]: {
    title: 'AlloyDB for Postgre SQL Cluster',
    description: 'AlloyDB for Postgre SQL Cluster.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_POSTGRE_SQL_INSTANCE]: {
    title: 'AlloyDB for Postgre SQL Instance',
    description: 'AlloyDB for Postgre SQL Instance.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_POSTGRE_SQL_CONNECTION]: {
    title: 'AlloyDB for Postgre SQL Connection',
    description: 'AlloyDB for Postgre SQL Connection.',
    defaultsToDisabled: false,
  },
  [IngestionSources.PROJECT_HAS_ALLOYDB_SERVICE_RELATIONSHIP]: {
    title: 'Project AlloyDB Service Has Relationship',
    description: 'Build Project AlloyDB Service Has Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.PROJECT_HAS_ALLOYDB_CLUSTER_RELATIONSHIP]: {
    title: 'Project AlloyDB Cluster Has Relationship',
    description: 'Build Project AlloyDB Cluster Has Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_INSTANCE_USES_CLUSTER_RELATIONSHIP]: {
    title: 'AlloyDB Instance Cluster Uses Relationship',
    description: 'Build AlloyDB Instance Cluster Uses Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_CLUSTER_HAS_BACKUP_RELATIONSHIP]: {
    title: 'AlloyDB Instance Backup Has Relationship',
    description: 'Build AlloyDB Backup Cluster Has  Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_INSTANCE_HAS_CONNECTION_RELATIONSHIP]: {
    title: 'AlloyDB Instance Connection Has  Relationship',
    description: 'Build AlloyDB Instance Connection Has Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_CLUSTER_USES_KMS_KEY_RELATIONSHIP]: {
    title: 'AlloyDB Cluster Kms Key Uses Relationship',
    description: 'Build AlloyDB Cluster Kms Key Has Relationship.',
    defaultsToDisabled: false,
  },
};

// IAM Permissions
export const AlloyDBPermissions = {
  STEP_ALLOYDB_POSTGRE_SQL_CLUSTER: ['alloydb.clusters.list'],
  STEP_ALLOYDB_POSTGRE_SQL_INSTANCE: ['alloydb.instances.list'],
  STEP_ALLOYDB_POSTGRE_SQL_CONNECTION: ['alloydb.instances.connect'],
  STEP_ALLOYDB_POSTGRE_SQL_BACKUP: ['alloydb.backups.get'],
};
