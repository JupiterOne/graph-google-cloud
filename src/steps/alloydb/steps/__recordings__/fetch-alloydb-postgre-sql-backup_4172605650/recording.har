{
  "log": {
    "_recordingName": "fetch-alloydb-postgre-sql-backup",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 709,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/9.4.1"
            },
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gl-node/18.18.2"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "709"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 293,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 1177,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 1177,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 11 Apr 2024 11:57:52 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 390,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-04-11T11:57:51.995Z",
        "time": 216,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 216
        }
      },
      {
        "_id": "854aaa9a9d5b8dba42c2e9162feb6eb5",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/7.0.1 gl-node/18.18.2"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/7.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "name": "host",
              "value": "alloydb.googleapis.com"
            }
          ],
          "headersSize": 1316,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://alloydb.googleapis.com/v1/projects/j1-gc-integration-dev-v3/locations/-/backups"
        },
        "response": {
          "bodySize": 1403,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 1403,
            "text": "{\"backups\":[{\"name\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/backups/contin-bkp-20240410-20-d94137c0-3e0d-405b-848d-7f591e65631e\",\"uid\":\"6a3a70d1-64b8-4a28-8da1-ef1f6388a619\",\"createTime\":\"2024-04-10T20:00:34.374893113Z\",\"updateTime\":\"2024-04-11T11:56:58.090787422Z\",\"state\":\"READY\",\"type\":\"CONTINUOUS\",\"clusterName\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/clusters/cluster-01\",\"reconciling\":false,\"encryptionInfo\":{\"encryptionType\":\"GOOGLE_DEFAULT_ENCRYPTION\"},\"sizeBytes\":\"534244\",\"clusterUid\":\"5f16156c-e137-4533-9e88-e4ef13908c08\",\"expiryTime\":\"2024-04-25T20:00:35.489778Z\",\"databaseVersion\":\"POSTGRES_15\"},{\"name\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/backups/contin-bkp-20240409-20-d3023018-e7ee-4462-8ee5-a13f950faa46\",\"uid\":\"c91cb7ee-19bb-420b-a992-c6cc96c446f2\",\"createTime\":\"2024-04-09T20:00:34.879375105Z\",\"updateTime\":\"2024-04-11T11:56:58.090625879Z\",\"state\":\"READY\",\"type\":\"CONTINUOUS\",\"clusterName\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/clusters/cluster-01\",\"reconciling\":false,\"encryptionInfo\":{\"encryptionType\":\"GOOGLE_DEFAULT_ENCRYPTION\"},\"sizeBytes\":\"585177\",\"clusterUid\":\"5f16156c-e137-4533-9e88-e4ef13908c08\",\"expiryTime\":\"2024-04-24T20:00:36.141154Z\",\"databaseVersion\":\"POSTGRES_15\"},{\"name\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/backups/contin-bkp-20240408-20-f626cb6b-a239-4f1f-bde7-11c049c3634d\",\"uid\":\"3d353f2e-cde0-417e-8e90-c61fc648b4e6\",\"createTime\":\"2024-04-08T20:00:43.449708845Z\",\"updateTime\":\"2024-04-11T11:56:58.116888336Z\",\"state\":\"READY\",\"type\":\"CONTINUOUS\",\"clusterName\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/clusters/cluster-01\",\"reconciling\":false,\"encryptionInfo\":{\"encryptionType\":\"GOOGLE_DEFAULT_ENCRYPTION\"},\"sizeBytes\":\"37390255\",\"clusterUid\":\"5f16156c-e137-4533-9e88-e4ef13908c08\",\"expiryTime\":\"2024-04-23T20:00:44.687844Z\",\"databaseVersion\":\"POSTGRES_15\"},{\"name\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/backups/contin-bkp-20240410-15-0f75bc45-e5cc-470d-a272-1c1c8150c06a\",\"uid\":\"dbbd820c-4bec-4442-bd7f-efc88d187e10\",\"createTime\":\"2024-04-10T15:00:34.466099456Z\",\"updateTime\":\"2024-04-11T11:56:58.209225424Z\",\"state\":\"READY\",\"type\":\"CONTINUOUS\",\"clusterName\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/clusters/cluster-05\",\"reconciling\":false,\"encryptionConfig\":{\"kmsKeyName\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/keyRings/test-Keyring-by-metron/cryptoKeys/High\"},\"encryptionInfo\":{\"encryptionType\":\"CUSTOMER_MANAGED_ENCRYPTION\",\"kmsKeyVersions\":[\"projects/j1-gc-integration-dev-v3/locations/us-east4/keyRings/test-Keyring-by-metron/cryptoKeys/High/cryptoKeyVersions/1\"]},\"sizeBytes\":\"37812684\",\"clusterUid\":\"2d2051af-2261-4a36-9701-080bf296ee2b\",\"expiryTime\":\"2024-04-25T15:00:35.799439Z\",\"databaseVersion\":\"POSTGRES_15\"},{\"name\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/backups/backupid-1\",\"uid\":\"95fe403a-1c34-4de9-9159-50e4c86df250\",\"createTime\":\"2024-04-08T13:09:48.695673731Z\",\"updateTime\":\"2024-04-11T11:56:58.237225958Z\",\"state\":\"READY\",\"type\":\"ON_DEMAND\",\"clusterName\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/clusters/cluster-01\",\"reconciling\":false,\"encryptionInfo\":{\"encryptionType\":\"GOOGLE_DEFAULT_ENCRYPTION\"},\"sizeBytes\":\"37352798\",\"clusterUid\":\"5f16156c-e137-4533-9e88-e4ef13908c08\",\"expiryTime\":\"2025-04-08T13:09:49.961981Z\",\"databaseVersion\":\"POSTGRES_15\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 11 Apr 2024 11:57:55 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 367,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-04-11T11:57:52.220Z",
        "time": 3041,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 3041
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
