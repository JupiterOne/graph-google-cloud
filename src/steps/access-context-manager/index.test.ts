import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { fetchAccessPolicies } from '.';
import { IntegrationConfig } from '../..';
import { integrationConfig } from '../../../test/config';
import { setupGoogleCloudRecording } from '../../../test/recording';
import { ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_ACCESS_POLICY } from './constants';

describe('#fetchAccessContextManagerAccessPolicies', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchAccessContextManagerAccessPolicies',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchAccessPolicies(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_ACCESS_POLICY,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['AccessPolicy'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_access_context_manager_access_policy' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          title: { type: 'string' },
          etag: { type: 'string' },
        },
      },
    });
  });
});

describe('#fetchAccessContextManagerAccessLevels', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchAccessContextManagerAccessPolicies',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchAccessPolicies(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_ACCESS_POLICY,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['AccessPolicy'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_access_context_manager_access_policy' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          title: { type: 'string' },
          etag: { type: 'string' },
        },
      },
    });
  });
});
