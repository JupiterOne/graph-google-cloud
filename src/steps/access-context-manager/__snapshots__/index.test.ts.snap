// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchAccessContextManagerAccessLevels should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "AccessPolicy",
      ],
      "_key": "accessPolicies/368605946207",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "etag": "64186ca0c82b01b4",
            "name": "accessPolicies/368605946207",
            "parent": "organizations/958457776463",
            "title": "default policy",
          },
        },
      ],
      "_type": "google_access_context_manager_access_policy",
      "createdOn": undefined,
      "displayName": "accessPolicies/368605946207",
      "etag": "64186ca0c82b01b4",
      "name": "accessPolicies/368605946207",
      "title": "default policy",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_access_context_manager_access_policy",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;

exports[`#fetchAccessContextManagerAccessPolicies should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "AccessPolicy",
      ],
      "_key": "accessPolicies/368605946207",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "etag": "64186ca0c82b01b4",
            "name": "accessPolicies/368605946207",
            "parent": "organizations/958457776463",
            "title": "default policy",
          },
        },
      ],
      "_type": "google_access_context_manager_access_policy",
      "createdOn": undefined,
      "displayName": "accessPolicies/368605946207",
      "etag": "64186ca0c82b01b4",
      "name": "accessPolicies/368605946207",
      "title": "default policy",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_access_context_manager_access_policy",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
